src/vis/vis.js

```diff
@@ -261,34 +259,20 @@ var VisModel = Backbone.Model.extend({
     this.settings.set(settings);
   },
 
-  _onMapInstanceCreated: function () {
-    this._analysisPoller.reset();
-    this._analysisCollection.each(function (analysisModel) {
-      analysisModel.unbind('change:status', this._onAnalysisStatusChanged, this);
-      if (analysisModel.url() && !analysisModel.isDone()) {
-        this._analysisPoller.poll(analysisModel);
-        this.trackLoadingObject(analysisModel);
-        analysisModel.bind('change:status', this._onAnalysisStatusChanged, this);
-      }
-    }, this);
+  _restartAnalysisPolling: function () {
+    this._analysisPoller.resetAnalysisNodes(this._getAnalysisNodeModels());
   },
 
-  _onAnalysisStatusChanged: function (analysisModel) {
-    if (analysisModel.isDone()) {
-      this.untrackLoadingObject(analysisModel);
-      if (this._isAnalysisSourceOfLayerOrDataview(analysisModel)) {
-        this.reload();
-      }
-    }
+  _getAnalysisNodeModels: function () {
+    return this._analysisCollection.models;
   },
 
   /**
    * Check if an Analysis node is the source of a layer or a dataview.
    */
   _isAnalysisSourceOfLayerOrDataview: function (analysisModel) {
-    var isAnalysisLinkedToLayer = this._isAnalysisLinkedToLayer(analysisModel);
-    var isAnalysisLinkedToDataview = this._dataviewsCollection.isAnalysisLinkedToDataview(analysisModel);
-    return isAnalysisLinkedToLayer || isAnalysisLinkedToDataview;
+    return this._layersCollection.isAnalysisLinkedToAnyLayer(analysisModel) ||
```

Comment:
I think we can use:

```javascript 
_isAnalysisSourceOfLayerOrDataview: function (analysisModel) {
  return analysisModel.isSourceOfAnyModel();
}
```

Reply:
Maybe `isAnalysisLinkedToAnyLayer` is not longer needed

Reply:
:sparkles: :sparkles: :sparkles:

Good catch!